pandas/tests/indexes/timedeltas/test_construction.py

```diff
@@ -5,11 +5,74 @@
 
 import pandas as pd
 import pandas.util.testing as tm
-from pandas import TimedeltaIndex, timedelta_range, to_timedelta
+from pandas import TimedeltaIndex, timedelta_range, to_timedelta, Timedelta
 
 
 class TestTimedeltaIndex(object):
 
+    def test_infer_from_tdi(self):
+        # GH#23539
+        # fast-path for inferring a frequency if the passed data already
+        #  has one
+        tdi = pd.timedelta_range('1 second', periods=10**7, freq='1s')
```

Comment:
we do not have any existing tests for this?

Reply:
nope